File: examples/tutorial-java/src/main/java/tutorial_java/service/TwitterService.java
package tutorial_java.service;

import java.util.List;
import java.util.Map;

import com.google.appengine.api.datastore.Transaction;

import org.slim3.datastore.Datastore;
import org.slim3.util.BeanUtil;

import tutorial_java.meta.TweetMeta;
import tutorial_java.model.Tweet;

public class TwitterService {

    private TweetMeta t = new TweetMeta();

    public Tweet tweet(Map<String, Object> input) {
        Tweet tweet = new Tweet();
        BeanUtil.copy(input, tweet);
        Transaction tx = Datastore.beginTransaction();
        Datastore.put(tweet);
        tx.commit();
        return tweet;
    }

    public List<Tweet> getTweetList() {
        return Datastore.query(t).sort(t.createdDate.desc).asList();
    }
}

File: examples/tutorial-scalate/src/main/scala/tutorial_scalate/service/TwitterService.scala
package tutorial_scalate.service

import java.util.List
import java.util.Map

import org.slim3.datastore.Datastore
import org.slim3.util.BeanUtil

import tutorial_scalate.meta.TweetMeta
import tutorial_scalate.model.Tweet

class TwitterService {
  private val t = new TweetMeta

  def tweet(input: Map[String, AnyRef]): Tweet = {
    val tweet = new Tweet
    BeanUtil.copy(input, tweet)
    val tx = Datastore.beginTransaction()
    Datastore.put(tweet)
    tx.commit()
    tweet
  }

  def getTweetList: List[Tweet] = {
    Datastore.query(t).sort(t.createdDate.desc).asList
  }
}
